src/scalefactors.hxx

```diff
@@ -62,5 +65,179 @@ auto iso(auto &df, const std::string &pt, const std::string &eta,
+
+vsElectrons                         | Value | Bit (value used in the config)
+------------------------------------|-------|-------
+no ID selection (takes every tau)   |  0    | -
+VVVLoose                            |  1    | 1
+VVLoose                             |  2    | 2
+VLoose                              |  4    | 3
+Loose                               |  8    | 4
+Medium                              |  16   | 5
+Tight                               |  32   | 6
+VTight                              |  64   | 7
+VVTight                             |  128  | 8
+
+vsMuons                             | Value | Bit (value used in the config)
+------------------------------------|-------|-------
+no ID selection (takes every tau)   |  0    | -
+VLoose                              |  1    | 1
+Loose                               |  2    | 2
+Medium                              |  4    | 3
+Tight                               |  8    | 4
+ * @param df The input dataframe
+ * @param pt tau pt
+ * @param eta tau eta
+ * @param decayMode decay mode of the tau
+ * @param genMatch column with genmatch values (from prompt e, prompt mu,
+ * tau->e, tau->mu, had. tau)
+ * @param selectedDMs list of allowed decay modes for which a scale factor
+ * should be calculated
+ * @param wp working point of the ID cut
+ * @param variation id for the variation of the scale factor "sf" for nominal
+ * and "systup"/"systdown" the up/down variation
+ * @param sf_dependence "pt", "dm" or "eta" based scale factors
+ * @param id_output name of the id scale factor column
+ * @param sf_file path to the file with the tau scale factors
+ * @param sf_name name of the tau id scale factor
+ * @return a new dataframe containing the new column
+ */
+auto id(auto &df, const std::string &pt, const std::string &eta,
+        const std::string &decayMode, const std::string &genMatch,
+        const std::vector<int> &selectedDMs, const std::string &wp,
+        const std::string &variation, const std::string &sf_dependence,
+        const std::string &id_output, const std::string &sf_file,
+        const std::string &sf_name) {
+
+    Logger::get("TauIDsf")->debug("Setting up functions for tau id sf");
+    Logger::get("TauIDsf")->debug("ID - Name {}", sf_name);
+    auto evaluator = correction::CorrectionSet::from_file(sf_file)->at(sf_name);
+    auto idSF_calculator = [evaluator, wp, variation, sf_dependence,
+                            selectedDMs, sf_name](
+                               const float &pt, const float &eta,
+                               const int &decayMode, const UChar_t &genMatch) {
+        // only calculate SFs for allowed tau decay modes
+        if (std::find(selectedDMs.begin(), selectedDMs.end(), decayMode) !=
+            selectedDMs.end()) {
+            double sf = 1.;
+            // differenciate between vsJet and vsEle/vsMu IDs due to different
+            // input parameters
+            if (sf_name == "DeepTau2017v2p1VSjet") {
```

Comment:
i would suggest spliiting into two functions, one for vsjet and one for vsele and vsmu, so keep this separation clear and remove the need to run an if statement for every single sf calculation

Reply:
I first had it in different functions and then changed it to one function because I thought this is easier to understand :D
But I can change it back.

Reply:
Done